state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -748,6 +789,23 @@ pub mod pallet {
 					Self::recycle_channel(&mut used_weight, address);
 				}
 			}
+
+			let mut expired_transactions = Vec::new();
+
+			for (account, tx_id, expire_block) in TaintedTransactions::<T, I>::iter() {
+				if expire_block <= now {
```

Comment:
I don't like having to iterate through all of the TaintedTransactions just to see if any have expired (99% of the time this will return nothing). It would be good to think of a better way to do this.

Reply:
We could introduce another storage item that also saves `ExpiryBlock => Vec<(account, tx_id)>` and then we change the TaintedTransactionMap to `AcountId, TxID => ()`. With this, we can remove the entry directly and don't need to iterate over all TaintedTransactions. As long no one is spamming, I don't see this would be a problem, but it's true that this could open up some kind of attack vector where any broker account could add a million tainted TX. I think we should avoid introducing any parameter that could affect the execution of a hook and can be manipulated from the outside.